File: php/create_file.php
#!/usr/bin/env php
<?php

if ($argc < 2) {
    echo "Usage: just create <filename>\n";
    exit(1);
}

$filename = $argv[1];
$rootDir = getcwd();
$srcDir = $rootDir . DIRECTORY_SEPARATOR . 'src';
$templateDir = $srcDir . DIRECTORY_SEPARATOR . 'template';
$testDir = $rootDir . DIRECTORY_SEPARATOR . 'Tests' . DIRECTORY_SEPARATOR . 'Unit';

// Get template contents
$srcTemplateFilename = $templateDir . DIRECTORY_SEPARATOR . "Source.php";
$srcTemplateContents = file_get_contents($srcTemplateFilename);
$testTemplateFilename = $templateDir . DIRECTORY_SEPARATOR . "Test.php";
$testTemplateContents = file_get_contents($testTemplateFilename);

// Create the source file and write template content
$srcFile = $srcDir . DIRECTORY_SEPARATOR . $filename . ".php";
$handle = fopen($srcFile, 'w');
fwrite($handle, $srcTemplateContents);
fclose($handle);

// Create the test file, replace $filename string and write the new content
$testFile = $testDir . DIRECTORY_SEPARATOR . $filename . "Test.php";
$handle = fopen($testFile, 'w');
$newContent = str_replace('$filename', $filename, $testTemplateContents);
fwrite($handle, $newContent);
fclose($handle);

echo "File $filename and its corresponding test created successfully from template src/template\n";

File: python/create_file.py
#!/usr/bin/env python3
import os
import shutil
import sys

filename_arg = sys.argv[1]
filename = filename_arg+".py"

# Get path of this script and add the filename argument
script_dir = os.path.dirname(os.path.realpath(__file__))
src_path = os.path.join(script_dir, 'src', filename)
test_path = os.path.join(script_dir, 'src', '__tests__', "test_"+filename)
template_src = os.path.join(script_dir, 'template', 'src_file.py')
template_test = os.path.join(script_dir, 'template', 'test_file.py')

# Copy files from template dir and place the source file in src, and test file in src/__tests__
shutil.copy(template_src, src_path)
shutil.copy(template_test, test_path)

# Then replace "filename" with the argument that was passed
with open(test_path, 'r+') as file:
    file_contents = file.read()
    file_contents = file_contents.replace("filename", filename_arg)
    file.seek(0)
    file.write(file_contents)
    file.truncate()

print(f"Created {filename} in src and __tests__")
